awx/main/analytics/core.py

```diff
@@ -154,9 +154,11 @@ def gather(dest=None, module=None, subset=None, since=None, until=None, collecti
         until = None if until is None else min(until, _now)
         since = None if since is None else min(since, _now)
 
-        if since and not until:
-            # If `since` is explicit but not `until`, `since` should be used to calculate the 4-week limit
-            until = min(since + timedelta(weeks=4), _now)
+        if since:
+            # If `since` is explicit, the four-week limit should be calculated from `since`
+            if until and until > since + timedelta(weeks=4):
```

Comment:
We want this message to trigger even when `until` is not specified. i.e. if a user specifies `--since <10 years ago>` we will move the date to `since + 4weeks` and we should let them know.